src/joystick/SDL_gamepad.c

```diff
@@ -1105,29 +1105,31 @@ const char *SDL_GetGamepadStringForAxis(SDL_GamepadAxis axis)
 }
 
 static const char *map_StringForGamepadButton[] = {
-    "a",
-    "b",
-    "x",
-    "y",
-    "back",
-    "guide",
-    "start",
-    "leftstick",
-    "rightstick",
-    "leftshoulder",
-    "rightshoulder",
-    "dpup",
-    "dpdown",
-    "dpleft",
-    "dpright",
-    "misc1",
-    "paddle1",
-    "paddle2",
-    "paddle3",
-    "paddle4",
-    "touchpad"
+    "a", NULL,
+    "b", NULL,
+    "x", NULL,
+    "y", NULL,
```

Comment:
It would be tempting to use

```
"a", "s",
"b", "e",
"x", "w",
"y", "n",
```

but I think that would be wrong for Nintendo controllers when `SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS` is enabled?

Reply:
SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS use in the mappings is going to go away. Instead it will affect the face button style returned from the API. I think this is a good idea, we'll just do that as a part of that separate change.

Reply:
SDL3 will have to either ignore mappings that specify SDL_HINT_GAMECONTROLLER_USE_BUTTON_LABELS=1, or map `A` to `EAST` and so on when it sees one of those mappings, even though it would normally map `A` to `SOUTH`.